pkg/v1/tarball/image.go

```diff
@@ -220,6 +220,17 @@ type uncompressedLayerFromTarball struct {
 	filePath  string
 }
 
+// foreignUncompressedLayer implements partial.UncompressedLayer but returns
+// a custom descriptor.
+type foreignUncompressedLayer struct {
```

Comment:
I think this wrapper is actually not necessary anymore. You could add `Descriptor` to `uncompressedLayerFromTarball` and replace [these](https://github.com/google/go-containerregistry/blob/5340dbaba4cf3d6a62218976684c9cbfc1628d32/pkg/v1/partial/uncompressed.go#L45-L46) with a `v1.Descriptor`.

Reply:
The struct merging has issues as discussed over IM